test/python/notebooks/test_jupyter.ipynb

```diff
@@ -0,0 +1,259 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-29T17:38:20.201993Z",
+     "start_time": "2018-10-29T17:38:19.154330Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sys\n",
+    "import time\n",
+    "cwd = os.getcwd()\n",
+    "qiskit_dir = os.path.abspath(os.path.join(cwd,\"../../..\"))\n",
+    "sys.path.append(qiskit_dir)"
```

Comment:
I'm not really a fan of this. Editing the system path is almost always the cause of more problems then it solves. I get that jupyter needs to build it's own ipython kernel to work so it adds another layer of complexity to the search path. But I'm a bit surprised that installing jupyter in the same python environment as the system installed qiskit python env doesn't make it discoverable. Like how does any pip installed package get discovered by jupyter if that doesn't work?

That being said if you don't think there is a way to get this working without editing the system path I'm fine with leaving this in.

Reply:
Qiskit is not installed on the test machines.  It is just being run from the directory.  As such, I cannot just import qiskit, and need to add the upper level dir to the path to get jupyter to find the qiskit module.  This was fixed in e54c8d5.

Reply:
Oh, then that's the bug, we should be installing qiskit in the python environment even for testing (in develop mode probably, so it just links to the source). I agree this is needed then until that's fixed.

Reply:
Have you tried using `Path` object to make it cross-platform?
Like:
```python
import os
from pathlib import Path
current_path = Path(__file__)
qiskit_dir = current_path.parent.parent.parent.absolute()
sys.path.append(qiskit_dir)
```
if `__file__` doesn't exist ion a Jupyter context, just use: `os.getcwd()`

Reply:
It already works on all the platforms, save for the AppVeyor issue we need to work around, so I am not quite sure why you are bringing that up?

Reply:
Nah, just thought this was not working on Windows because the code was using *nix style paths: `../../` ... but if it's working, then forget it.